Analysis/Tasks/PWGHF/HFCandidateCreatorXicc.cxx

```diff
@@ -0,0 +1,271 @@
+                         std::sqrt(impactParameter0.getSigmaY2()), std::sqrt(impactParameter1.getSigmaY2()),
+                         xicCand.globalIndex(), trackpion.globalIndex(),
+                         hfFlag);
+      } // if on selected Xicc
+    }   // loop over candidates
+  }     // end of process
+};      //end of struct
+
+/// Extends the base table with expression columns.
+struct HfCandidateCreatorXiccExpressions {
+  Spawns<aod::HfCandXiccExt> rowCandidateXicc;
+  void init(InitContext const&) {}
+};
+
+/// Performs MC matching.
+struct HfCandidateCreatorXiccMc {
+  Produces<aod::HfCandXiccMCRec> rowMCMatchRec;
+  Produces<aod::HfCandXiccMCGen> rowMCMatchGen;
+
+  void process(aod::HfCandXicc const& candidates,
+               aod::HfCandProng3,
+               aod::BigTracksMC const& tracks,
+               aod::McParticles const& particlesMC)
+  {
+    int indexRec = -1;
+    int8_t sign = 0;
+    int8_t flag = 0;
+    int8_t origin = 0;
+
+    // Match reconstructed candidates.
+    for (auto& candidate : candidates) {
+      //Printf("New rec. candidate");
+      flag = 0;
+      origin = 0;
+
+      auto xicCand = candidate.index0();
+      auto arrayDaughters = array{xicCand.index0_as<aod::BigTracksMC>(),
+                                  xicCand.index1_as<aod::BigTracksMC>(),
+                                  xicCand.index2_as<aod::BigTracksMC>(),
+                                  candidate.index1_as<aod::BigTracksMC>()};
+
+      // Ξcc±± → p± K∓ π± π±
+      //Printf("Checking Ξcc±± → p± K∓ π± π±");
+      indexRec = RecoDecay::getMatchedMCRec(particlesMC, arrayDaughters, pdg::Code::kXiCCPlusPlus, array{+kProton, -kKPlus, +kPiPlus, +kPiPlus}, true, &sign, 3);
+      if (indexRec > -1) {
+        flag = 1 << DecayType::XiccToXicPi;
+      }
+
+      // Check whether the particle is non-prompt (from a b quark).
+      if (flag != 0) {
+        auto particle = particlesMC.iteratorAt(indexRec);
+        origin = (RecoDecay::getMother(particlesMC, particle, kBottom, true) > -1 ? OriginType::NonPrompt : OriginType::Prompt);
+      }
+
+      rowMCMatchRec(flag, origin);
+    }
+
+    // Match generated particles.
+    for (auto& particle : particlesMC) {
+      //Printf("New gen. candidate");
+      flag = 0;
+      origin = 0;
+      // Xicc → Xic + π+
+      if (RecoDecay::isMatchedMCGen(particlesMC, particle, pdg::Code::kXiCCPlusPlus, array{int(pdg::Code::kXiCPlus), +kPiPlus}, true)) {
+        // Match Xic -> pKπ
+        std::vector<int> arrDaughter;
+        RecoDecay::getDaughters(particlesMC, particle, &arrDaughter, array{int(pdg::Code::kXiCPlus)}, 1);
+        auto XicCandMC = particlesMC.iteratorAt(arrDaughter[0]);
+        //Printf("Checking Ξc± → p± K∓ π±");
+        if (RecoDecay::isMatchedMCGen(particlesMC, XicCandMC, int(pdg::Code::kXiCPlus), array{+kProton, -kKPlus, +kPiPlus}, true, &sign)) {
+          flag = sign * (1 << DecayType::XiccToXicPi);
```

Comment:
`sign` should correspond to the one returned by the first `isMatchedMCGen` and the second `isMatchedMCGen` should be called with `acceptAntiParticles = false` to guarantee the same sign of the Ξc daughter.

Reply:
hi @vkucera I need some clarification on this point, this is why I haven't considered it yet. thanks.

Reply:
Hi @vkucera, I understood the point (finally 👍 ). I will fix it accordingly